ClinGen/clincoded
# Connects #427 kl homozygous

**Changes:**
- Add a new property "genotype" (non-required) in schema individual.json for storing item "Mode of Inheritance" in interface.
- Add handling codes in family_curation.js, individual_curtion.js and curator.js to:
  1.  add a pull-down list in interfaces for the new field titled Mode of Inheritance;
  2.  enable/disable variant fields;
  3.  control error messages;
  4.  control submission;
  5.  display genotype (Mode of Inheritance) in view pages;
  6.  assign genotype value to pre-R4 product data and let user edit later.

**Test for Creating/Editing/Viewing Individual**
- Create a GDM and add PMID 123;
- Add an individual, Ind001, select No at proband, expect to see Mode of Inheritance field in variant panel like:
![summary](https://cloud.githubusercontent.com/assets/9206696/12728285/f07969d8-c8d5-11e5-89bc-0bc4ea9045d4.png)
- Select "Dominant", expect to see required VariationID and Other description fields for one variant like:
![summary](https://cloud.githubusercontent.com/assets/9206696/12728438/b8f04ec2-c8d6-11e5-9f3f-9e3cb1b9e348.png)
- Change Mode of Inheritance to "Compound Heterozygous", expect to see fields for 2 variants.
![summary](https://cloud.githubusercontent.com/assets/9206696/12728515/017c5cd0-c8d7-11e5-91c0-ce8260782550.png)
- Click Save without entering data in any variant field, expect to see error messages under each of the fields as:
![summary](https://cloud.githubusercontent.com/assets/9206696/12728602/79b2b9c4-c8d7-11e5-9bfc-ea25b2c431ec.png)
- Go back to "No Selection" at Mode of Inheritance, expect to see all error messages cleared.
- Select "Compound Heterozygous" and click Save again, after error messages displayed, change Mode of Inheritance value to "Homozygous Recessive", expect to see fields for the one variant only (with error message).
![summary](https://cloud.githubusercontent.com/assets/9206696/12729071/c70080c4-c8d9-11e5-857e-e8e616fe056f.png)
- Enter variation id 111, expect to see error messages cleared and Other description field disabled as:
![summary](https://cloud.githubusercontent.com/assets/9206696/12729163/337068b4-c8da-11e5-9531-d1109a69e4f6.png)
- Click Save and expect save the individual successfully.
- Go to curation central page and click View Ind001. Expect to see variant info as:
![summary](https://cloud.githubusercontent.com/assets/9206696/12729315/eebdef56-c8da-11e5-8808-db0aa43330f6.png)
- Go to curation central page and click Edit Ind001.
- Change Mode of Inheritance value to "Compound Heterozygous" and enter "NM_000527.4:c.1467C>G" in Other description field of the 2nd variant, click Save.
- Expect edit the individual successfully.
- Go to curation central and view the individual, expect to see new genotype (Mode of Inheritance) and variant data as:
![summary](https://cloud.githubusercontent.com/assets/9206696/12729577/5e3ba11a-c8dc-11e5-9b0a-fc15685e9edd.png)
- Edit Ind001 again and select "No Selection" at Mode of Inheritance, click Save.
- View Ind001 and expect to see "No variant associated." in Associated Variant(s) panel.
![summary](https://cloud.githubusercontent.com/assets/9206696/12729862/af13d886-c8dd-11e5-9bf5-971109a59ec6.png)

**Test for Creating/Editing/Viewing Family**
- Add family Family010 to the PMID, select genotype value "Dominant", enter variant 222, then expect to trigger proband fields shown as:
![summary](https://cloud.githubusercontent.com/assets/9206696/12730042/aaac3454-c8de-11e5-806e-767b09f3440d.png)
- Click Save without enter proband data, expect to see error messages like:
![summary](https://cloud.githubusercontent.com/assets/9206696/12731846/a7f5794c-c8e7-11e5-8998-149f6196ae70.png)
- Change Mode of Inheritance to "No Selection", expect to see no variant field displayed and clear error messages.
- Select "Dominant", associate variant 222, name proband Ind011 and enter a Orphanet id, and lick Save, expect to save the family successfully.
- View Family010 and expect to see genotype value (Dominant) and variant id (222).
![summary](https://cloud.githubusercontent.com/assets/9206696/12731601/8eb34730-c8e6-11e5-97f0-66262b2b8055.png)
- View Ind011, expect to see the same genotype value and variation id.
- Edit family010, change Mode of Inheritance value to "No Selection", click Save, expect to see error message as:
![screen shot 2016-02-05 at 12 45 52 pm](https://cloud.githubusercontent.com/assets/9206696/12858685/71fe9284-cc06-11e5-861d-7fa773790323.png)
- Change  to any other option, expect to see the error messages cleared.
- Select option "Compound Heterozygous", expect to see data fields for the 2nd variant.
- Click Save without entering any thing in the fields, expect to see error messages.
![summary](https://cloud.githubusercontent.com/assets/9206696/12731742/2c955a1a-c8e7-11e5-97ab-ff4820964b75.png)
- Enter variation id 333 for the 2nd variant and click Save again, expect to save the family successfully.
- View Family010 and expect to see "Compound Heterozygous", variationID 222 and variationID 333.
- View Ind011 (proband) and expect to see the same genotype and variation IDs.

**End of Test**
